internal/runner/tools/install.go

```diff
@@ -88,18 +90,32 @@ func InstallBinaries(layer *configv1alpha1.TerraformLayer, repo *configv1alpha1.
 		}
 	}()
 
-	terraformVersion := configv1alpha1.GetTerraformVersion(repo, layer)
-	terraformVersion, err = detect(binaryPath, "terraform", terraformVersion)
-	if err != nil {
-		return nil, err
+	var baseExec BaseExec
+	var baseExecVersion string
+	if configv1alpha1.GetTerraformEnabled(repo, layer) {
+		baseExecVersion, err = detect(binaryPath, "terraform", configv1alpha1.GetTerraformVersion(repo, layer))
+		if err != nil {
+			return nil, err
+		}
+		baseExec = &tf.Terraform{
+			ExecPath: filepath.Join(binaryPath, "Terraform", baseExecVersion, "terraform"),
+		}
+	} else if configv1alpha1.GetOpenTofuEnabled(repo, layer) {
+		baseExecVersion, err = detect(binaryPath, "tofu", configv1alpha1.GetOpenTofuVersion(repo, layer))
+		if err != nil {
+			return nil, err
+		}
+		baseExec = &ot.OpenTofu{
+			ExecPath: filepath.Join(binaryPath, "OpenTofu", baseExecVersion, "tofu"),
+		}
+	} else {
+		return nil, errors.New("Please enable either Terraform or OpenTofu in the repository or layer configuration")
```

Comment:
If an user forgets to enable Terraform or OpenTofu in their CRDs, it would be better to be informed before a runner is spawned.

It can be more intuitive to the user (searching through runner logs is not obvious), maybe it can be checked by the Layer controller or the Run controller ?

Reply:
Yes you're right, I will add a controller log and and event in the related resource

Reply:
Okay. In the TerraformLayer controller or the TerraformRun controller ?

Reply:
I was thinking adding this to the layer controller